### ISSUE TITLE
Use abs_path instead of absoluteURI for Request-URI* in HTTP client ?

### ISSUE THREAD

**Issue by jlcanela** [dream11/zio-http#237]
Would it be possible to use abs_path instead of absoluteURI for *5.1.2 Request-URI* in HTTP client ?
Reference: https://www.w3.org/Protocols/rfc2616/rfc2616-sec5.html

Context: 
* using zio-http Client.request("http://localhost:8000/") 
* error 404 when testing a service not supporting absoluteURI

Current behavior: 
* use absoluteURI to provide Request-URI in request
```
127.0.0.1 - - [25/May/2021 22:01:35] code 404, message File not found
127.0.0.1 - - [25/May/2021 22:01:35] "GET http://localhost:8000/ HTTP/1.1" 404 -
```

Expected behavior:
* use abs_path to provide Request-URI in request
```
127.0.0.1 - - [25/May/2021 21:45:01] "GET / HTTP/1.1" 200 -
```

EncodeRequest.encodeRequest is computing uri as req.url.asString and so always provide absoluteURI instead of abs_path when not using proxy

Some HTTP servers do not support absoluteURI although they should be. 
Example of non compliant server:  **python3 -m http.server**

**Comment by jlcanela** [comment]
I suggest something like https://github.com/jlcanela/zio-http/commit/4ee25a1b14c5f6613e8b4662e6065d6f42ed6f08

**Comment by jlcanela** [comment]
Note that if using proxy settings https://github.com/dream11/zio-http/issues/101 it’s mandatory to use absoluteURI in the request.

**Comment by ShrutiVerma97** [comment]
@jlcanela can you help me understand the use-case where `absoluteURI` doesn't work to get a better idea.

**Comment by jlcanela** [comment]
Sure, when you have an absoluteURI the http request contains the following: 
```
GET http://localhost:8000/ HTTP/1.1
```

When you use abs_path  the http request contains the following:
```
GET / HTTP/1.1
```

When you use zio-http as web server server absoluteURI is properly managed, unfortunately that’s not the case for all http server:  **python3 -m http.server**, the local python server expects the URI to be the file path to serve and cannot file the file to server when the uri is http://localhost:8000/.

Note that curl is using abs_path:
```
 curl -v http://localhost:8000/
*   Trying 127.0.0.1:8000...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8000 (#0)
> GET / HTTP/1.1
> Host: localhost:8000
> User-Agent: curl/7.68.0
> Accept: */*
```

Note that when using a proxy config (zio-http client does not support proxy config yet) it’s mandatory to use absoluteURI and not abs_path.

**Comment by tusharmath** [comment]
Fixed in #239 Thank you @jlcanela !
